knative-operator/pkg/controller/knativekafka/rbacproxytransforms.go

```diff
@@ -0,0 +1,50 @@
+package knativekafka
+
+import (
+	"context"
+	"errors"
+
+	mf "github.com/manifestival/manifestival"
+	operatorv1alpha1 "github.com/openshift-knative/serverless-operator/knative-operator/pkg/apis/operator/v1alpha1"
+	"github.com/openshift-knative/serverless-operator/openshift-knative-operator/pkg/monitoring"
+	"k8s.io/apimachinery/pkg/util/sets"
+	eventingv1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	kafkaChannelComponents = []string{"kafka-ch-controller", "kafka-ch-dispatcher", "kafka-webhook"}
+	kafkaSourceComponents  = []string{"kafka-controller-manager"}
+)
+
+func addRBACProxySupportToManifest(instance *operatorv1alpha1.KnativeKafka, components []string) (*mf.Manifest, error) {
+	proxyManifest := mf.Manifest{}
+	// Only create the roles needed for the deployment service accounts as Prometheus has already
+	// the rights needed due to eventing that is assumed to be installed.
+	for _, c := range components {
```

Comment:
could we use `kafkaDeployments`?  (looking here: https://github.com/openshift-knative/serverless-operator/blob/6f9d2ca6d6a8a61b251315d2d326b330094dcdca/openshift-knative-operator/pkg/monitoring/monitoring_eventing.go#L36)

Reply:
This is different part. The Kafka setup is done in the knative-operator. This needs to remain as is because it deals with the Eventing components like broker-ingress, broker-filter etc.